src/tsolvers/lrasolver/LRAModel.cc

```diff
@@ -90,24 +88,32 @@ void LRAModel::clear() {
 
 int LRAModel::backtrackLevel() { return bound_limits.size() - 1; }
 
-LABoundRef LRAModel::readLBoundRef(LVRef v) const { return int_lbounds[getVarId(v)].last(); }
+
+LABoundRef LRAModel::readLBoundRef(LVRef v) const { assert(hasLBound(v)); return int_lbounds[getVarId(v)].last(); }
 const LABound& LRAModel::readLBound(const LVRef &v) const { return bs[readLBoundRef(v)]; }
-LABoundRef LRAModel::readUBoundRef(LVRef v) const { return int_ubounds[getVarId(v)].last(); }
+LABoundRef LRAModel::readUBoundRef(LVRef v) const { assert(hasUBound(v)); return int_ubounds[getVarId(v)].last(); }
 const LABound& LRAModel::readUBound(const LVRef &v) const { return bs[readUBoundRef(v)]; }
 void LRAModel::pushBacktrackPoint()      { bound_limits.push(bound_trace.size()); }
 void LRAModel::popBacktrackPoint() { popBounds(); bound_limits.pop(); }; // Returns the decision if the backtrack point had a decision
 int  LRAModel::getBacktrackSize() const { return bound_limits.size(); }
 
-bool LRAModel::isEquality(LVRef v) const { return bs[int_lbounds[getVarId(v)].last()].getIdx().x+1 == bs[int_ubounds[getVarId(v)].last()].getIdx().x && !Lb(v).isInf() && !Ub(v).isInf() && Lb(v) == Ub(v); }
+//bool LRAModel::isEquality(LVRef v) const { return hasLBound(v) && hasUBound(v) && bs[int_lbounds[getVarId(v)].last()].getIdx().x+1 == bs[int_ubounds[getVarId(v)].last()].getIdx().x && !Lb(v).isInf() && !Ub(v).isInf() && Lb(v) == Ub(v); }
```

Comment:
Is there a way of getting the equalities now?  I think this might be important in the LIA implementation.  Can be like this for now, though, given that the LIA implementation is still to materialise...

Reply:
I belie I have commented it out, because it was not used anywhere in current implementation. I am bringing it back.
It should return true when there is both upper bound and lower bound and they have the same value, correct? I am not sure what does this check represents `bs[int_lbounds[getVarId(v)].last()].getIdx().x+1 == bs[int_ubounds[getVarId(v)].last()].getIdx().x`

Reply:
> It should return true when there is both upper bound and lower bound and they have the same value, correct?

Yes.

> bs[int_lbounds[getVarId(v)].last()].getIdx().x+1 == bs[int_ubounds[getVarId(v)].last()].getIdx().x

It should represent that the lower bound is the same as upper bound, and it's correctness depends on the exact way how bounds are ordered in the bound lists.  I don't remember how well it's been tested though.

Reply:
OK, so, why isn't this check enough?
``` Lb(v) == Ub(v)```
Is there a situation when these two checks would return different result?